### PR TITLE
imu

### PR DESCRIPTION
**Pull request by matthew-reynolds**
Slapped together imu driver.

A few TODOs. Mainly, make the imu driver safer to access from the rt thread, and less hardcoded values. The latter is not that important though. make it work, not make it nice.

### CODE REVIEW

File: qpup_hw/src/qpup_hw.cpp [hunk 1]
```diff
@@ -12,10 +12,22 @@ bool QPUPHW::init(ros::NodeHandle & /*root_nh*/, ros::NodeHandle &robot_hw_nh) {
     registerCommandInterfacesAndTransmissions(joint_name);
   }
 
+  // Register ImuSensorHandle
+  // TODO(mreynolds): hardcoding
+  imu_sensor_interface_.registerHandle(hardware_interface::ImuSensorHandle(
+      "imu",
+      "base_link",
+      imu_states_["imu"].orientation, imu_states_["imu"].orientation_covariance,
```
Comment (matthew-reynolds) [inline qpup_hw/src/qpup_hw.cpp]:
```suggestion
      imu_states_["imu"].orientation,
      imu_states_["imu"].orientation_covariance,
```

File: qpup_hw/CMakeLists.txt [hunk 2]
```diff
@@ -5,8 +5,10 @@ find_package(catkin REQUIRED COMPONENTS
         hardware_interface
         joint_limits_interface
         roscpp
+        tf2
         transmission_interface
         )
+find_package(Threads REQUIRED)
```
Comment (wmmc88) [inline qpup_hw/CMakeLists.txt:11]: lol does it not compile without this?

File: qpup_hw/include/qpup_hw/navx/AHRS.h [hunk 3]
```diff
@@ -0,0 +1,205 @@
+/*
+ * AHRS.h
+ *
+ *  Created on: Jul 30, 2015
+ *      Author: Scott
```
Comment (wmmc88) [inline qpup_hw/include/qpup_hw/navx/AHRS.h:5]: noice

**Review by wmmc88 — APPROVED** [review]

File: qpup_hw/CMakeLists.txt (hunk 2 above)
Comment (matthew-reynolds) [reply qpup_hw/CMakeLists.txt:11]: Nope, need -pthreads. In most ros stuff, people just get it transitively lol

**Review by dawsonhemphill — APPROVED** [review]

**Pull request merged**